stl/inc/ranges

```diff
@@ -4748,6 +4748,566 @@ namespace ranges {
+            requires (!_Simple_view<_Vw>) {
+            // clang-format on
+            return _Iterator<false>{this, _RANGES begin(_Range)};
+        }
+
+        _NODISCARD constexpr auto begin() const
+            noexcept(noexcept(_RANGES begin(_Range))
+                     && is_nothrow_move_constructible_v<iterator_t<const _Vw>>) /* strengthened */
+            requires forward_range<const _Vw> {
+            return _Iterator<true>{this, _RANGES begin(_Range)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() noexcept(noexcept(_RANGES distance(_Range))
+            && noexcept(_RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) /* strengthened */
+            requires (!_Simple_view<_Vw>) {
+            // clang-format on
+            if constexpr (common_range<_Vw> && sized_range<_Vw>) {
+                const auto _Missing = (_Count - _RANGES distance(_Range) % _Count) % _Count;
+                return _Iterator<false>{this, _RANGES end(_Range), _Missing};
+            } else if constexpr (common_range<_Vw> && !bidirectional_range<_Vw>) {
+                return _Iterator<false>{this, _RANGES end(_Range)};
+            } else {
+                return default_sentinel;
+            }
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() const
+            noexcept(noexcept(_RANGES distance(_Range)) && noexcept(_RANGES begin(_Range))
+                     && is_nothrow_move_constructible_v<iterator_t<const _Vw>>) /* strengthened */
+            requires forward_range<const _Vw> {
+            // clang-format on
+            if constexpr (common_range<const _Vw> && sized_range<const _Vw>) {
+                const auto _Missing = (_Count - _RANGES distance(_Range) % _Count) % _Count;
+                return _Iterator<true>{this, _RANGES end(_Range), _Missing};
+            } else if constexpr (common_range<const _Vw> && !bidirectional_range<const _Vw>) {
+                return _Iterator<true>{this, _RANGES end(_Range)};
+            } else {
+                return default_sentinel;
+            }
+        }
+
+        _NODISCARD constexpr auto size() noexcept(noexcept(_RANGES distance(_Range))) /* strengthened */
+            requires sized_range<_Vw> {
+            return _To_unsigned_like(_Div_ceil(_RANGES distance(_Range), _Count));
+        }
+
+        _NODISCARD constexpr auto size() const noexcept(noexcept(_RANGES distance(_Range))) /* strengthened */
+            requires sized_range<const _Vw> {
+            return _To_unsigned_like(_Div_ceil(_RANGES distance(_Range), _Count));
+        }
+    };
+
+    template <class _Rng>
+    chunk_view(_Rng&&, range_difference_t<_Rng>) -> chunk_view<views::all_t<_Rng>>;
+
+    template <class _Vw>
+    inline constexpr bool enable_borrowed_range<chunk_view<_Vw>> = enable_borrowed_range<_Vw>&& forward_range<_Vw>;

```

Comment:
These are swapped from the original in the standard but I'm okay with that since `enable_borrowed_range` is assumedly much easier to check for falsity than `forward_range`

Reply:
really? the `&&` doesn't short circuit even in non-concept locations?

Reply:
Correct - `conjunction` provides short-circuiting behavior but plain `&&` won't (for variable template scenarios like this; evaluation within a `constexpr` function is different).

Fully worked example: https://godbolt.org/z/9hhbfYovE

```cpp
#include <type_traits>
using namespace std;

template <typename T> struct always_false_also_error_for_pointer : false_type {
    static_assert(!is_pointer_v<T>);
};

template <typename A> struct test_short_circuit {
#ifdef USE_DISJUNCTION
    static constexpr bool value = disjunction_v<is_pointer<A>, always_false_also_error_for_pointer<A>>;
#else
    static constexpr bool value = is_pointer_v<A> || always_false_also_error_for_pointer<A>::value;
#endif
};

static_assert(test_short_circuit<int*>::value);

int main() {}
```